src/routes/TraderForm/Form/index.tsx

```diff
@@ -9,18 +17,45 @@ import { NotionalSwap } from './NotionalSwap';
 import { SubmitButton } from './SubmitButton';
 
 export const Form: React.FunctionComponent = () => {
+  const dispatch = useAppDispatch();
+
+  const fixedRateInfo = useAppSelector(selectFixedRateInfo);
+  const variableRateInfo = useAppSelector(selectVariableRateInfo);
+  const mode = useAppSelector(selectMode);
+
+  const handleOnChange = useCallback(
+    (value?: 'fixed' | 'variable') => {
+      if (!value) {
+        return;
+      }
+      dispatch(
+        setModeAction({
+          value,
+        }),
+      );
+    },
+    [dispatch],
+  );
+
   // TODO: move the mode to the redux store
-  const [mode, setMode] = useState<'fixed' | 'variable'>('fixed');
   return (
     <FormBox>
       <Typography colorToken="lavenderWeb" typographyToken="primaryHeader3Black">
         SWAP Rates
       </Typography>
-      <NotionalSwap fixedRate={5.49} mode={mode} variableRate={2.49} onSwap={setMode} />
+      <NotionalSwap
```

Comment:
Right now when fixedRateInfo, variableRateInfo, mode change all 5 components re-render even though some of them don't need to re-render.

I think I made initial arch. design mistake. 

It is better if we remove these props from here (from NotionalSwap) and move the logic inside the NotionalSwap component. That way if those values change in state we avoid re-render of all 5 components in the Form and only render the NotionalSwap component. Can you do such refactor?

Reply:
[957310c](https://github.com/Voltz-Protocol/voltz-ui/pull/571/commits/957310cc8a0b03d2680b71d8f8ec5e98803f7150)